src/snowflake/snowpark/_internal/analyzer/expression.py

```diff
@@ -405,6 +409,14 @@ def __init__(
         if nanosecond is not None:
             self.values_dict["NANOSECOND"] = nanosecond
 
+        if self._ast is None and _emit_ast:
+            expr = proto.Expr()
+            ast = with_src_position(expr.sp_interval)
+            # Set the AST values based on the values_dict.
+            for k, v in self.values_dict.items():
+                getattr(ast, k.lower()).value = v
```

Comment:
let's try to avoid indirection via getattr and use the values explicitly. It's much easier to read and reason about :)